concordium-consensus/src/Concordium/ProtocolUpdate/P5/ProtocolP6.hs

```diff
@@ -0,0 +1,113 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeFamilies #-}
+
+-- FIXME: This is currently a dummy update, and the details, including the update hash, need to be resolved before release.
+-- TODO: Consider moving the state changes that are made here into the state migration. [This cannot be done after release.]
+
+-- |This module implements the P5.ProtocolP6 protocol update.
+-- This protocol update is valid at protocol version P6, and updates
+-- to protocol version P6.
+-- The block state is preserved across the update.
+--
+-- This produces a new 'RegenesisDataP6' using the 'GDP6Regenesis' constructor,
+-- as follows:
+--
+-- * 'genesisCore':
+--
+--     * 'genesisTime' is the timestamp of the last finalized block of the previous chain.
+--     * 'genesisEpochDuration' is 1 hour.
+--     * 'genesisSignatureThreshold' is 2/3.
+--
+-- * 'genesisFirstGenesis' is either:
+--
+--     * the hash of the genesis block of the previous chain, if it is a 'GDP5Initial'; or
+--     * the 'genesisFirstGenesis' value of the genesis block of the previous chain, if it
+--       is a 'GDP5Regenesis'.
+--
+-- * 'genesisPreviousGenesis' is the hash of the previous genesis block.
+--
+-- * 'genesisTerminalBlock' is the hash of the last finalized block of the previous chain.
+--
+-- * 'genesisStateHash' and 'genesisNewState' are the hash and (V0) serialized state of the
+--   new genesis block, which are derived from the block state of the last finalized block of
+--   the previous chain by applying the following changes:
+--
+--     * The 'SeedState' is updated with:
+--
+--         * 'epochLength' is unchanged;
+--         * 'epoch' is @0@;
+--         * 'currentLeadershipElectionNonce' is the SHA256 hash of (@"Regenesis" <> encode (updatedNonce oldSeedState)@); and
+--         * 'updatedNonce' is the same as 'currentLeadershipElectionNonce'.
```

Comment:
I cannot see this being done in the code below.

Reply:
copy and pase err